configure.ac

```
@@ -4098,8 +4098,10 @@ AS_VAR_IF([have_libffi], [yes], [
 # See https://github.com/python/cpython/issues/125206 for more details.
 #
 AC_CACHE_CHECK([libffi has complex type support], [ac_cv_ffi_complex_double_supported],
-[ac_save_cc="$CC"
-CC="$CC -lffi"
+[WITH_SAVE_ENV([
+ CFLAGS="$LIBFFI_CFLAGS $CFLAGS"
```

Let's use `CPPFLAGS` instead; we've had `CFLAGS` fail mysteriously on FreeBSD earlier.

```suggestion
 CPPFLAGS="$LIBFFI_CFLAGS $CPPFLAGS"
```

---

CPPFLAGS are flags for the C preprocessor?

---

Yes, correct; the C preprocessor will use `CPPFLAGS` (but not `CFLAGS`), and the compiler will use both `CFLAGS` and `CPPFLAGS`. Sometimes Autoconf will run parts of a check only with the pre-processor. We've had issues with this before. I think that for the following `AC_RUN_IFELSE` macro, `CFLAGS` should be fine, but I prefer to play safe and use `CPPFLAGS`.

---

The error https://github.com/python/cpython/issues/125206#issuecomment-2443158799 may be caused by missing *linker* flags, LDFLAGS?

---

I'll do, JFR above usage was taken from the libffi test.